parse.go

```diff
@@ -289,6 +291,71 @@ func (p *parser) parsePartial() (node, error) {
 	return &partialNode{t.val}, nil
 }
 
+// parseSection parses a test_Value block. It is assumed that the next read should
+// return a t_section token.
+func (p *parser) parseTest() (node, error) {
+	t := p.read()
+	if t.typ != tokenIdentifier {
+		return nil, p.errorf(t, "unexpected token %s", t)
+	}
+	if next := p.read(); next.typ != tokenLeftDelim {
+		return nil, p.errorf(next, "unexpected token %s", next)
+	}
+	i := p.read()
+	if i.typ != tokenIdentifier {
+		return nil, p.errorf(i, "unexpected token %s", i)
+	}
+	if next := p.read(); next.typ != tokenRightDelim {
+		return nil, p.errorf(next, "unexpected token %s", next)
+	}
+
+	v := p.read()
+	if v.typ != tokenText {
+		return nil, p.errorf(v, "unexpected token %s", v)
+	}
+
+	if next := p.read(); next.typ != tokenRightDelim {
+		return nil, p.errorf(next, "unexpected token %s", next)
+	}
+
+	var (
```

Comment:
Can line 312 - Line 346 (or Line 350) be factored out to something like parserSessionInternal()? Looks like both parseSession() and parseTest() share the same logic this part.

Reply:
The conditional inside the case is different I'll think about if I can come up with a clever way to keep the code readable and factor it out.

Reply:
Ack.